documentation/inventory_file_v2.md

```diff
@@ -68,18 +68,14 @@ groups:
 
 ### Target object
 
-You can specify a target with the string representation of a URI, or as a hash
-with any of the following fields:
+Specify a target under the `targets` key. You can specify a target's name in the following ways:
+- Use an array with the string representation of the target's Universal Resource Identifier (URI).
+- Use a hash and specify a human-readable `name` or the `uri` of the target with an optional `alias`.
 
-| Key | Description | Type |
-| --- | ----------- | ---- |
-| `alias` | A unique alias to refer to the target. Optional. | `String` |
-| `config` | The configuration for the target. Optional. | `Hash` |
-| `facts` | The facts for the target. Optional. | `Hash` |
-| `features` | The features for the target. Optional. | `Array[String]`
-| `name` | A human-readable name for a target.<br> **Required** when specifying a target using a hash. Optional if using `uri`. | `String` |
-| `uri` | The URI of the target.<br> **Required** when specifying a target using a hash. Optional if using `name`. | `String` |
-| `vars` | The variables for the target. Optional. | `Hash` |
+If you don't specify a `name` for your target, Bolt sets the name to the `uri`.
+If you set a `name` and no `uri`, and the transport you're using supports
+hostname configuration, Bolt uses the `host` configuration. If the transport
+you're using does not support hostname configuration, you must set a `uri`.
```

Comment:
I think it'd be helpful to explain what it uses `host` configuration for. Bolt will use the target's `uri` to establish a connection to the target, and falls back to `host` if the `uri` is not specified.

Reply:
Maybe explain the "lookup order".

AFAIK: If all keys are present, `uri` overwrites what is specified in `host` and `host` overwrites what is specified in `name` when trying to connect. `uri` > `host` > `name`. If one of those is not present, Bolt starts evaluating from left to right.

A proper and easy to understand sentence for this would be hard, but necessary, IMHO.

Reply:
Bolt actually does not use the target's `name` to establish connections, it's only used to reference the target from the inventory. I think maybe that's where some confusion lies, since you can 1) name a target and 2) specify the target's host. The `uri` key does both of those.

| | Name a target | Connect to a target |
| --- | --- | --- |
| **Higher precedence** | `name` | `uri` |
| **Lower precedence** | `uri` | `host` |

In the case of naming a target, either a `uri` or `name` is required, otherwise Bolt will error when you load inventory. In the case of connecting to a target, either `uri` or `host` is required, otherwise Bolt will error when you try connecting to the target.

Reply:
I've added some language around `uri` and `name` to make this a little clearer. I also messed with the examples a bit.